design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+  running `go tool nm authenticator` and verifying that it has symbols named
+   *_Cfunc__goboringcrypto_*
+   
+More info on the required changes can be found [here](https://github.com/cyberark/conjur-authn-k8s-client/pull/97)
+
+### Non-FIPS compliant image
+
+We should decide if we still want to release a version of the `conjur-authn-k8s-client`
+that is not FIPS compliant (i.e that uses `golang` as the base image). Let's
+look into some pros and cons of both approaches.
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that Releasing both
+versions will bring to the development team.
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
```

Comment:
We created a performance baseline that we should follow.

Reply:
Link?

Reply:
@hilagross I don't think it is relevant here. That performance test was on decomposed follower. It has nothing to do with the performance of the authenticator client.

We should write a performance test and run it once on the current `master` image and once on the FIPS compliant one.

Reply:
@sigalsax link is in Slack

@orenbm following our talk today I agree, we should understand what and how to run performance tests and test them for k8s-authn-client specifically. I believe it should be basic and simple, not necessarily automated for this task.

Reply:
In Secretless we run some long-time stability tests that also validate the authenticator client stability.

We do not have anything that tests how quickly the access token is retrieved, however.